q2_fmt/_peds.py

```diff
@@ -170,24 +186,43 @@ def _check_for_time_column(metadata, time_column):
     try:
         num_timepoints = metadata[time_column].dropna().unique().size
     except Exception as e:
-        raise KeyError('There was an error finding %s in the metadata'
-                       % time_column) from e
+        if time_column == metadata.index.name:
+            raise KeyError('The `--p-time-column` input provided was the same'
+                           ' as the index of the metadata. `--p-time-column`'
+                           ' can not be the same as the index of metadata:'
+                           ' `%s`' % time_column) from e
+        else:
+            raise KeyError('There was an error finding the provided'
+                           ' `--p-time-column`: `%s` in the metadata'
+                           % time_column) from e
     return num_timepoints
 
 
-def _check_time_column_numeric(metadata, time_column):
+def _check_time_column_numeric(column_properties, time_column):
```

Comment:
Hey @cherman2, I think you re-use iterations of this helper method enough times that it would be worth a slight refactor (post-2023.5, since I know we're right up against a deadline on this). I'd re-write something like this:
```
def _check_column_type(column_properties, column, column_type):
    try:
        assert column_properties[column].type == column_type
    except AssertionError as e:
        raise AssertionError('Non-%s values found in `--p-%s-column`. Please make sure the column selected contains the correct MetadataColumn type. Column with non-%s values that was selected: %s' % (column_type, column, column_type, column)) from e
```
I'm not sure if the syntax of the AssertionError message is exactly correct, but this is mainly just to show an example for how I'd refactor all of these helpers.